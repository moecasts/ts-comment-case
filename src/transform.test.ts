import { readFileSync } from 'fs';
import { join } from 'path';
import { describe, expect, test } from 'vitest';

import {
  CommentCase,
  transform,
  transformCommentIntoJSDocStyle,
  transformCommentIntoSingleLineStyle,
} from './transform';

const code = readFileSync(join(__dirname, '__materials__/code.ts'), 'utf-8');
const code2 = readFileSync(join(__dirname, '__materials__/code-2.ts'), 'utf-8');

code;
code2;

describe('transform', () => {
  test('transformCommentIntoSingleLineStyle', () => {
    const comments = [
      '/** multi line comment A  b */',

      '// single line comment',

      `/** multi line comment A  b
      */`,

      `
/**
 * multi line comment A
 * multi line comment B
 *      multi line comment C
 */`,
    ];

    comments.forEach((comment) => {
      const result = transformCommentIntoSingleLineStyle(comment);
      expect(result).toMatchSnapshot();
    });
  });

  test('transformCommentIntoJSDocStyle', () => {
    const comments = ['// single line comment', '/** single line comment */'];

    comments.forEach((comment) => {
      const result = transformCommentIntoJSDocStyle(comment);
      expect(result).toMatchSnapshot();
    });
  });

  test('transform comment into single line style', () => {
    const result = transform(code, { commentCase: CommentCase.Single });
    expect(result).toMatchSnapshot();
  });

  test('transform comment into JSDoc style', () => {
    const result = transform(code, { commentCase: CommentCase.JSDoc });
    expect(result).toMatchSnapshot();
  });
});
