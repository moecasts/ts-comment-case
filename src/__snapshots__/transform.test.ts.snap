// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`transform > transform comment into JSDoc style 1`] = `
"/**
 * single line comment
 */"
`;

exports[`transform > transform comment into JSDoc style 2`] = `"/** single line comment */"`;

exports[`transform > transform comment into JSDoc style 3`] = `
"/**
 * This is an interface called Test.
 */
/**
 * JSDoc comment 1
 * @description this is a JSDoc comment description
 */
/**
 * This is an interface called Test.
 * @description this is a description.
 * "double quote"
 */
export interface Test {
  /**
   * this is an interface prop called name.
   */
  name: string;

  /**
   * this is an interface prop called age.
   * @description this is a description.
   */
  age: number;
}

/**
 * JSDoc comment 1
 * @description this is a JSDoc comment description
 */
/**
 * JSDoc comment 2
 * @description this is a JSDoc comment description
 */
export const jsdoc = () => {
  /**
   * Single line comment
   */
  const singleLineComment = 'singleLineComment'; // inline comment

  /*
   * Block line comment
   */
  const blockLineComment = 'blockLineComment'; // inline comment

  /**
   * inner JSDoc comment
   *
   * @description this is a JSDoc comment description
   */
  const innerJSDocComment = 'innerJSDocComment'; // inline comment

  /**
   * Multi line comment
   * Multi line comment
   */
  console.log({
    singleLineComment,
    blockLineComment,
    innerJSDocComment,
  });
};
"
`;

exports[`transform > transform comment into single line style 1`] = `
"// This is an interface called Test.
// JSDoc comment 1
// @description this is a JSDoc comment description
// This is an interface called Test.
// @description this is a description.
// "double quote"
export interface Test {
  // this is an interface prop called name.
  name: string;

  // this is an interface prop called age.
  // @description this is a description.
  age: number;
}

// JSDoc comment 1
// @description this is a JSDoc comment description
// JSDoc comment 2
// @description this is a JSDoc comment description
export const jsdoc = () => {
  // Single line comment
  const singleLineComment = 'singleLineComment'; // inline comment

  // Block line comment
  const blockLineComment = 'blockLineComment'; // inline comment

  // inner JSDoc comment
  // @description this is a JSDoc comment description
  const innerJSDocComment = 'innerJSDocComment'; // inline comment

  // Multi line comment
  // Multi line comment
  console.log({
    singleLineComment,
    blockLineComment,
    innerJSDocComment,
  });
};
"
`;

exports[`transform > transformCommentIntoSingleLineStyle 1`] = `"// multi line comment A  b"`;

exports[`transform > transformCommentIntoSingleLineStyle 2`] = `"// single line comment"`;

exports[`transform > transformCommentIntoSingleLineStyle 3`] = `"// multi line comment A  b"`;

exports[`transform > transformCommentIntoSingleLineStyle 4`] = `
"// multi line comment A
// multi line comment B
//      multi line comment C"
`;
